pyccel/parser/semantic.py

```diff
@@ -884,27 +881,21 @@ def _create_variable(self, name, dtype, rhs, d_lhs):
 
     def _ensure_target(self, rhs, d_lhs):
         """ Function using data about the new lhs to determine
-        whether the lhs is a pointer and the rhs is a target
+        whether the lhs is an alias and the rhs is a target
         """
-        if isinstance(rhs, NumpyTranspose) and rhs.internal_var.allocatable:
-            d_lhs['allocatable'] = False
-            d_lhs['is_pointer' ] = True
-            d_lhs['is_stack_array'] = False
 
+        if isinstance(rhs, NumpyTranspose) and rhs.internal_var.on_heap:
+            d_lhs['memory_handling'] = 'alias'
             rhs.internal_var.is_target = True
+
         if isinstance(rhs, Variable) and rhs.is_ndarray:
-            d_lhs['allocatable'] = False
-            d_lhs['is_pointer' ] = True
-            d_lhs['is_stack_array'] = False
+            d_lhs['memory_handling'] = 'alias'
+            rhs.is_target = not rhs.is_alias
 
-            rhs.is_target = not rhs.is_pointer
         if isinstance(rhs, IndexedElement) and rhs.rank > 0 and \
-                (getattr(rhs.base, 'is_ndarray', False) or getattr(rhs.base, 'is_pointer', False)):
-            d_lhs['allocatable'] = False
-            d_lhs['is_pointer' ] = True
-            d_lhs['is_stack_array'] = False
-
-            rhs.base.is_target = not rhs.base.is_pointer
+                (getattr(rhs.base, 'is_ndarray', False) or getattr(rhs.base, 'memory_handling', False) == 'alias'):
```

Comment:
Instead of `getattr(rhs.base, 'memory_handling', False) == 'alias'):` would `getattr(rhs.base, 'is_alias', False)` be clearer?